File: 0x01-lockboxes/0-lockboxes.py
#!/usr/bin/python3
"""LockBoxes"""


def canUnlockAll(boxes):
    """Function to check if all boxes can be unlocked
       Boxes is a list of lists
    """

    unlocked = {0}
    length = len(boxes)
    for index, value in enumerate(boxes):
        if not value:
            continue
        for key in value:
            if (key < length) and (key not in unlocked) and (key != index):
                unlocked.add(key)
    return len(unlocked) != length

File: 0x01-lockboxes/0-lockboxes.js
export default function canUnlockAll(boxes) {
    if (!(Array.isArray(boxes))) {
        return false;
    }
    const length = boxes.length;
    const unlocked = [0];

    for (let i = 0; i < length; i++) {
        if (Array.isArray(boxes[i])) {
            for (let key of boxes[i]) {
                if ((key < length) && (key !== i) && (!(key in unlocked))) {
                    unlocked.push(key);
                }
            }
        }
    }
    if (unlocked.length !== length) {
        return false
    }
    return true;
}